src/module/item/spellcasting-entry/document.ts

```diff
@@ -329,18 +329,6 @@ class SpellcastingEntryPF2e<TParent extends ActorPF2e | null = ActorPF2e | null>
 
         await super._preUpdate(changed, options, user);
     }
-
-    /* -------------------------------------------- */
-    /*  Deprecations and Compatibility              */
-    /* -------------------------------------------- */
-
-    /**
-     * To prevent (or delay) console spam, will send out a deprecation notice in a later release
-     * @deprecated
-     */
-    getSpellData(): Promise<SpellcastingSheetData> {
-        return this.getSheetData();
-    }
```

Comment:
I've actually been unsure about this one (I've mentioned it before). Not only did it never have a deprecation warning, but I think this signature is actually more accurate than getSheetData(). This function is used by modules to determine what spells are available to cast at what levels.

I think we should keep this one around, and then consider having a way to acquire "active spells" from a collection that getSheetData() uses. Then we can probably kill this function without remorse.

Reply:
No belly aching out of you!

Reply:
This is currently the only way to acquire certain mechanically relevant information, and its known public api that never had a warning or any communication that its going away.

We need to continue work on collections anyways for future features.

Reply:
ok fine!